### PR TITLE
add proto interface for descheduler

### PR DESCRIPTION
**Pull request by Garrybest**
**What type of PR is this?**
/kind api-change

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes part of #1262 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/estimator/pb/types.go [hunk 1]
```diff
@@ -47,3 +49,45 @@ type MaxAvailableReplicasResponse struct {
...
+type MaxUnschedulableReplicasRequest struct {
+	// Cluster represents the cluster name.
+	// +required
+	Cluster string `json:"cluster" protobuf:"bytes,1,opt,name=cluster"`
+	// Resource represents the Kubernetes resource to be propagated.
+	// +required
+	Resource ObjectReference `json:"resource" protobuf:"bytes,2,opt,name=resource"`
+	// Resource represents the Kubernetes resource to be propagated.
+	// +optional
+	UnscheduableThreshold time.Duration `json:"unscheduableThreshold,omitempty" protobuf:"varint,3,opt,name=unscheduableThreshold,casttype=time.Duration"`
```
Comment (RainbowMango) [inline pkg/estimator/pb/types.go:63]:
```suggestion
	UnschedulableThreshold time.Duration `json:"unschedulableThreshold,omitempty" protobuf:"varint,3,opt,name=unschedulableThreshold,casttype=time.Duration"`
```
s/Unscheduable/Unschedulable/

Comment (RainbowMango) [reply pkg/estimator/pb/types.go:63]: The comments of `UnschedulableThreshold` are not correct. (Don't know why lint ignored this. :( )

File: pkg/estimator/pb/types.go [hunk 2]
```diff
@@ -47,3 +49,45 @@ type MaxAvailableReplicasResponse struct {
 	// +required
 	MaxReplicas int32 `json:"maxReplicas" protobuf:"varint,1,opt,name=maxReplicas"`
 }
+
+// MaxUnschedulableReplicasRequest represents the request that sent by gRPC client to calculate unschedulable replicas.
+type MaxUnschedulableReplicasRequest struct {
```
Comment (RainbowMango) [inline pkg/estimator/pb/types.go:54]: What does the `Max` mean? In my opinion, what the client request is `unschedulable` replicas, there is no notion of a maximum.

File: pkg/estimator/pb/types.go (hunk 1 above)
Comment (Garrybest) [reply pkg/estimator/pb/types.go:63]: Sorry, I alse made this mistake in proposal.

File: pkg/estimator/pb/types.go (hunk 2 above)
Comment (Garrybest) [reply pkg/estimator/pb/types.go:54]: Right.

File: pkg/estimator/pb/types.go [hunk 3]
```diff
@@ -47,3 +49,45 @@ type MaxAvailableReplicasResponse struct {
...
+// UnschedulableReplicasRequest represents the request that sent by gRPC client to calculate unschedulable replicas.
+type UnschedulableReplicasRequest struct {
+	// Cluster represents the cluster name.
+	// +required
+	Cluster string `json:"cluster" protobuf:"bytes,1,opt,name=cluster"`
+	// Resource represents the Kubernetes resource to be propagated.
+	// +required
+	Resource ObjectReference `json:"resource" protobuf:"bytes,2,opt,name=resource"`
+	// Resource represents the Kubernetes resource to be propagated.
+	// +optional
```
Comment (RainbowMango) [inline pkg/estimator/pb/types.go:62]: I guess you forgot update the comments :)

**Review by RainbowMango — COMMENTED** [review]
/lgtm
/approve

File: pkg/estimator/pb/types.go [hunk 4]
```diff
@@ -47,3 +49,46 @@ type MaxAvailableReplicasResponse struct {
...
+	// Name represents the name of the referent.
+	// +required
+	Name string `json:"name" protobuf:"bytes,4,opt,name=name"`
+}
+
+// UnschedulableReplicasResponse represents the response that sent by gRPC server to calculate unschedulable replicas.
+type UnschedulableReplicasResponse struct {
+	// UnschedulableReplicas represents the unschedulable replicas that the object contains.
+	// +required
+	UnschedulableReplicas int32 `json:"maxUnschedulableReplicas" protobuf:"varint,1,opt,name=maxUnschedulableReplicas"`
```
Comment (RainbowMango) [inline pkg/estimator/pb/types.go:93]: missed the json tag. Glad to see it is fixed by #1299

**Pull request merged**